core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java

```diff
@@ -294,13 +296,13 @@ public void close() {
                   }
                   remoteResults.onCompletion(null, ourSegments, localValue);
                } else {
+                  IntSet ourSegments = IntSets.mutableFrom(ch.getPrimarySegmentsForOwner(localAddress));
```

Comment:
Maybe we should have this a little closer to what it was before, but I like the new debug message. This way we don't copy the IntSet for the case when `segmentsToProcess` is null, which can happen during a rehash very easily.

```java
IntSet ourSegments;
if (segmentsToProcess != null) {
   ourSegments = IntSets.mutableFrom(ch.getPrimarySegmentsForOwner(localAddress));
   ourSegments.retainAll(segmentsToProcess);
} else {
   ourSegments = IntSets.from(ch.getPrimarySegmentsForOwner(localAddress));
}
getLog().debugf("Local cache topology changed from %d to %d, suspecting local segments %s",
   cacheTopology.getTopologyId(), newCacheTopology.getTopologyId(), ourSegments);
remoteResults.onSegmentsLost(ourSegments);
```

Reply:
I was thinking that this shouldn't happen too often because the topology doesn't change that often, but I'll look into using `LocalizedCacheTopology.getLocalReadSegments()` as well.